scripts/analyse-go-binary.py

```diff
@@ -0,0 +1,200 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+
+import re
+import sys
+import struct
+import logging
+import argparse
+from typing import List, Iterable, Optional
+
+import pefile
+from tqdm import tqdm
+
+from floss.results import StaticString, StringEncoding
+
+logger = logging.getLogger(__name__)
+
+
+final_size = 0
+
+MIN_STR_LEN = 4
+
+
+def find_longest_string(bindata):
+    """
+    Reference:- https://github.com/CarveSystems/gostringsr2/blob/master/gostringsr2/gostringsr2.py
+    """
+    off = 0
+    this_off = 0
+    longest_off = 0
+    longest_size = 0
+
+    binlength = len(bindata)
+    while off < binlength:
+        b = bindata[off : off + 2]
+        # Basically, terminate a "string" if 2 null bytes are seen. Seems to work for the most part.
+        if b == b"\x00\x00":
+            this_size = off - this_off
+            if this_size > 0:
+                if this_size > longest_size:
+                    longest_off = this_off
+                    longest_size = this_size
+            this_off = off + 2
+        else:
+            this_size = off - this_off
+            if this_size > 0:
+                if this_size > longest_size:
+                    longest_off = this_off
+                    longest_size = this_size
+        off += 2
+
+    if (off - this_off) > longest_size:
+        longest_off = this_off
+        longest_size = off - this_off
+
+    if longest_size > 0:
+        return (longest_off, longest_size)
+
+    return (None, 0)
+
+
+def extract_go_strings(
```

Comment:
let's keep the functionality separate and reuse the Go extractor code

Reply:
instead of working off the master branch, you can branch off / merge into the branch from #792